cli/root_test.go

```diff
@@ -16,8 +20,75 @@ import (
 	"github.com/coder/coder/cli/cliflag"
 	"github.com/coder/coder/cli/clitest"
 	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/testutil"
 )
 
+var updateGoldenFiles = flag.Bool("update", false, "update .golden files")
+
+//nolint:tparallel,paralleltest // These test sets env vars.
+func TestCommandHelp(t *testing.T) {
+	t.Parallel()
```

Comment:
You use `t.Setenv` in the subtests. Can't we not set envs during parallel tests?

Reply:
Nope, `t.Setenv` will panic after a call to `t.Parallel()`. It modifies the os env but restores it at the end of the test.

Reply:
Yes, apparently it does not panic if you use `t.Setenv` in a subtest and you have  `t.Parallel` on the root test.

`TestCommandHelp` sets `t.Parallel` on the first line, but subtests use `t.Setenv`. Do we need to remove `t.Parallel` from the root test?

Reply:
Huh?

This is from the Go implementation:

```go
func (t *T) Setenv(key, value string) {
	if t.isParallel {
		panic("testing: t.Setenv called after t.Parallel; cannot set environment variables in parallel tests")
	}
```

So if the subtest is parallel, it'll panic, no?

Reply:
Ah I think I misunderstood. It’s fine if the parent test has t.Parallel. Just means that starting it will not block other tests, but the subtests will.

Reply:
:+1: